src/test/ui/async-await/nested-async-calls.stderr

```diff
@@ -0,0 +1,42 @@
+error[E0277]: future cannot be sent between threads safely
+  --> $DIR/nested-async-calls.rs:26:5
+   |
+LL | fn require_send<T: Send>(_val: T) {}
+   |    ------------    ---- required by this bound in `require_send`
+...
+LL |     require_send(wrapped);
+   |     ^^^^^^^^^^^^ future returned by `first` is not `Send`
+   |
+   = help: within `main::Wrapper<impl std::future::Future>`, the trait `std::marker::Send` is not implemented for `*const ()`
+   = note: required because it appears within the type `third::{{closure}}#0::NotSend`
+note: future is not `Send` as this value is used across an await
+  --> $DIR/nested-async-calls.rs:17:5
+   |
+LL |     let _a: Outer;
+   |         -- has type `third::{{closure}}#0::Outer`
+LL |     dummy().await;
+   |     ^^^^^^^^^^^^^ await occurs here, with `_a` maybe used later
+LL | }
+   | - `_a` is later dropped here
+note: future is not `Send` as this value is used across an await
```

Comment:
This part is kind of a "stack trace" indicating how `third` wound up being invoked from `first`. I could certainly see that being useful -- but I think we could present it in a more concise way, and maybe clarify better for the user what is happening.

Reply:
i.e., maybe we can explain to them that this is effectively a "stack trace" or something

Reply:
One small tweak to this could be to have all the "future is not `Send` as this value is used across an await" to be in the same `note`.

Reply:
We can take inspiration from the `-Zexternal-macro-backtrace` output.